discovery/client.go

```diff
@@ -77,6 +78,26 @@ func NewClient(ctx context.Context, issuer string, opts ...ClientOpt) (*Client,
 		return nil, fmt.Errorf("error decoding provider metadata response: %v", err)
 	}
 
+	c.refresher = &refresher{
+		hc:      c.hc,
+		jwksuri: c.md.JWKSURI,
+		stop:    make(chan struct{}, 1),
+	}
+
+	if err := c.RefreshJWKS(ctx); err != nil {
+		return nil, fmt.Errorf("initial jwks fetch: %w", err)
+	}
+
+	if c.backgroundRefresh {
+		if c.backgroundRefreshInterval == 0 {
+			c.backgroundRefreshInterval = 15 * time.Minute
+		}
+		go c.refresher.RunRefreshRoutine(c.backgroundRefreshInterval)
+		runtime.SetFinalizer(c, func(c *Client) {
```

Comment:
Why not use the already passed in context for this?

Reply:
Contexts are usually request scoped, and I don't love long-lived contexts. They get annoying in that you have to make sure it's alive for the scope of the client, and then remember to correctly shut it down when the client is done. This removes needing to manually manage that, and avoids leaking routines. I don't think it really adds any cost either.